index.js

```diff
@@ -47,49 +49,84 @@ class Rewt {
     this.options = options;
     this._redisConn = options.redisConn;
     this._scripty = new Scripty(this._redisConn);
+    this._shouldCacheSecret = options.cacheSecret || false;
   }
 
   /**
    * Signs the given payload.
    *
    * @param {Object} payload The payload to sign.
-   * @param {Function} cb Node style callback.
    */
-  sign(payload, cb) {
-    this._getSecret((err, val) => {
-      if (err) {
-        cb(err);
-      } else {
-        jwt.sign(payload, val, {}, cb);
-      }
+  async sign(payload) {
+    const secret = await this._getSecret();
+    return new Promise((resolve, reject) => {
+      jwt.sign(payload, secret, {}, (err, val) => {
```

Comment:
Probably not worth pulling in `promise-callbacks`.promisify here, eh?

Reply:
Honestly, I don't think so - it also keeps the amount of runtime dependencies as minimal as possible.